pkg/model/hetznermodel/servers.go

```diff
@@ -33,6 +34,24 @@ type ServerGroupModelBuilder struct {
 var _ fi.ModelBuilder = &ServerGroupModelBuilder{}
 
 func (b *ServerGroupModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	var sshkeyTasks []*hetznertasks.SSHKey
+	for _, sshkey := range b.SSHPublicKeys {
```

Comment:
Nit: If you have two ServerGroupModelBuilders, with overlapping SSHPublicKeys, this may result in duplicate tasks.  So you _might_ want to consider keeping the dedicated SSHPublicKeyBuilder task, or using EnsureTask (which allows for duplicate creation if they are identical, but is a bit more hacky).  But not a blocker!

Reply:
The ServerGroupModelBuilder creates servers for all instance groups, so there is no need for using EnsureTask at the moment.